apps/src/applab/designMode.js

```diff
@@ -9,16 +9,36 @@ var elementLibrary = require('./designElements/library');
 var elementUtils = require('./designElements/elementUtils');
 var studioApp = require('../StudioApp').singleton;
 var KeyCodes = require('../constants').KeyCodes;
-var constants = require('./constants');
-var applabCommands = require('./commands');
+var applabConstants = require('./constants');
 var designMode = module.exports;
 var sanitizeHtml = require('./sanitizeHtml');
 var utils = require('../utils');
 var gridUtils = require('./gridUtils');
 var logToCloud = require('../logToCloud');
+var actions = require('./actions');
 
 var currentlyEditedElement = null;
-var currentScreenId = null;
+var ApplabMode = applabConstants.ApplabMode;
+
+/**
+ * Subscribe to state changes on the store.
+ * @param {!Store} store
+ */
+designMode.subscribeToRedux = function (store) {
+  var state = {};
+  store.subscribe(function () {
+    var lastState = state;
+    state = store.getState();
+
+    if (state.currentScreenId !== lastState.currentScreenId) {
+      onScreenChange(state.currentScreenId);
+    }
+
+    if (state.mode !== lastState.mode) {
+      onModeChange(state.mode);
+    }
+  });
+};
```

Comment:
One reason that having a bunch of subscribers that update UI vs. having a single root component that subscribes/connects is that it's not much less predictable what order things happen in.